src/common/net/fetch-client.ts

```diff
@@ -151,6 +193,61 @@ export class FetchHttpClient extends HttpClient implements HttpClientInterface {
 
     return new FetchHttpClientResponse(res);
   }
+
+  private async fetchRequestWithRetry(
+    url: string,
+    method: string,
+    body?: any,
+    headers?: RequestHeaders,
+  ): Promise<HttpClientResponseInterface> {
+    let response: HttpClientResponseInterface;
+    let retryAttempts = 1;
+
+    const makeRequest = async (): Promise<HttpClientResponseInterface> => {
+      let requestError: any = null;
+
+      try {
+        response = await this.fetchRequest(url, method, body, headers);
+      } catch (e) {
+        requestError = e;
+      }
+
+      if (this.shouldRetryRequest(requestError, retryAttempts)) {
+        retryAttempts++;
+        await this.sleep(retryAttempts);
+        return makeRequest();
+      }
+
+      if (requestError != null) {
+        throw requestError;
+      }
+
+      return response;
+    };
+
+    return makeRequest();
+  }
+
+  private shouldRetryRequest(requestError: any, retryAttempt: number): boolean {
+    if (retryAttempt > this.MAX_RETRY_ATTEMPTS) {
+      return false;
+    }
+
+    if (requestError != null) {
+      if (requestError instanceof TypeError) {
+        return true;
+      }
```

Comment:
What's an example of a TypeError we'd get here that we'd want to retry?

Reply:
The main errors we're trying to handle here are request timeouts that happen if a check or query take too long. One that came up recently was: `Unexpected error: TypeError: fetch failed` which happened because a 499 was returned after 5s.

Here's the full list of errors `fetch` throws a `TypeError` for: https://developer.mozilla.org/en-US/docs/Web/API/Window/fetch#exceptions